lib/SILGen/SILGenApply.cpp

```diff
@@ -4427,6 +4427,27 @@ class FixLifetimeDestroyCleanup : public Cleanup {
 #endif
   }
 };
+
+class EmitBreadcrumbCleanup : public Cleanup {
+  ExecutorBreadcrumb breadcrumb;
+
+public:
+  EmitBreadcrumbCleanup(ExecutorBreadcrumb &&breadcrumb)
+    : breadcrumb(std::move(breadcrumb)) {}
+
+  void emit(SILGenFunction &SGF, CleanupLocation l,
+            ForUnwind_t forUnwind) override {
+    breadcrumb.emit(SGF, l);
+  }
+
+  void dump(SILGenFunction &SGF) const override {
+#ifndef NDEBUG
+    llvm::errs() << "EmitBreadcrumbCleanup "
+                 << "State:" << getState()
+                 << "NeedsEmit:" << breadcrumb.needsEmit();
```

Comment:
Not the most important thing, but there's a missing space here before `NeedsEmit:`

Reply:
it's also suppose to have newlines and such, now that I've been debugging the clean-ups and relying on it 😄